### PR TITLE
Correctly propagate anchors for combined diacritics

### PR DESCRIPTION
**Pull request by madig**
In Cantarell, I combine some diacritics like `circumflexcomb` and `tildecomb` into `circumflexcomb_tildecomb` for some Vietnamese characters. The anchor propagation filter doesn't correctly propagate anchors for these composites. The resulting `ocircumflextilde` has its `top` anchor at the position in the `o` instead of the position in the `tildecomb.loclVIET`.

Here's a reference generated by Glyphs: [CantarellAnchorPropagation.zip](https://github.com/googlei18n/ufo2ft/files/2865170/CantarellAnchorPropagation.zip)

The test is constructed such that I test that the anchors I want are where I want them to be; Glyphs knows not to export `circumflexcomb.loclVIET`'s `top.viet` anchor to `circumflexcomb_tildecomb` because `tildecomb` is attached to `circumflexcomb.loclVIET`'s `top` anchor, but there's no built-in way to determine this in the UFO format. I have code that guesses what attaches to what and where based on anchor distance, used by our QA scripts, which I can probably apply to this, but it's a lot of code compared to the current propagation module...

### CODE REVIEW

**Comment by madig** [comment]
Cantarell's `DZcaron` doesn't get anchors propagated like in Glyphs. Not sure it would make any sense, but maybe I need to add it to the test anyway.

**Comment by madig** [comment]
Some of the propagations make no sense, although Glyphs does them, too.

bildschirmfoto vom 2019-02-16 00-24-54

I understand that the anchors from the base glyph are copied verbatim, unless a mark has an anchor with the same name, then that anchor is used. Here I constructed the `Oslash` from `O` and `slashlongcomb.case`, the latter has no new `center` anchor so `Oslash` is left with the original `center` anchor.

**Comment by madig** [comment]
Hm. ufoLib2's `Component` doesn't have a `bounds()` method.

**Comment by madig** [comment]
ugh wait. glyphSet isn't guaranteed to have the full set...

**Comment by madig** [comment]
Or is it?

**Comment by madig** [comment]
Ok, reading the comments, the purpose of `glyphSet` is to provide all necessary glyph objects, the glyphs to run the filter on are stored in `include`. So this code should be fine.

File: Lib/ufo2ft/filters/propagateAnchors.py [hunk 1]
```diff
@@ -77,6 +80,22 @@ def _propagate_glyph_anchors(glyphSet, composite, processed):
                 base_components.append(component)
                 anchor_names |= {a.name for a in glyph.anchors}
 
+    if mark_components and not base_components and _is_ligature_mark(composite):
+        # The composite is a mark that is composed of other marks (E.g.
+        # "circumflexcomb_tildecomb"). Promote the mark that is positioned closest
+        # to the origin to a base.
+        try:
+            component = _component_closest_to_origin(mark_components, glyphSet)
+        except Exception as e:
```
Comment (anthrotype) [inline Lib/ufo2ft/filters/propagateAnchors.py]: a bit too broad catch-all exception for my taste

**Comment by anthrotype** [comment]
I only looked at this briefly and I don't understand what the issue is, and why you have to "promote a mark that is closest to the origin to a base". I'll try again another time :)

File: Lib/ufo2ft/filters/propagateAnchors.py (hunk 1 above)
Comment (madig) [reply Lib/ufo2ft/filters/propagateAnchors.py]: But is that a problem? The purpose here is to catch everyhting and bubble it up with information about which glyph processing failed.

**Comment by madig** [comment]
The problem is that anchors are not propagated on glyphs that are composites of marks because none of them is considered a base of the other and the existing code wants a base to do anything.

In the test UFO, I constructed `ocircumflextilde` out of `o` and `circumflexcomb_tildecomb`, which is a composite of `circumflexcomb.loclVIET` and `tildeflexcomb.loclVIET`. For `circumflexcomb_tildecomb`, the code at https://github.com/googlei18n/ufo2ft/pull/316/files#diff-fe62cded474563610a1479bcbe3b9789R83 determines that we are looking at two marks and no bases, and that the glyph name suggests that the glyph is in fact a ligature (because it has a `_` in its name, but not at the beginning, which Glyphs uses to say "this is a non-export glyph"). Now we need to determine which mark should be the base for all the others, and I do that by determining which is visually the lowest (using the bounds of the components in the space of the composite).

**Comment by schriftgestalt** [comment]
> I do that by determining which is visually the lowest (using the bounds of the components in the space of the composite).

That is dangerous (the position of the components might be different in different masters) Just use the underscore anchor from the first component. And the next base anchor from the last component that has one (if you like to have mkmk on that combined mark). 

See https://github.com/googlefonts/glyphsLib/issues/794

**Pull request merged**